zerver/views/realm.py

```diff
@@ -352,9 +354,32 @@ def update_realm(
 
 @require_realm_owner
 @has_request_variables
-def deactivate_realm(request: HttpRequest, user: UserProfile) -> HttpResponse:
+def deactivate_realm(
+    request: HttpRequest,
+    user: UserProfile,
+    realm_deletion_in_minutes: Optional[int] = REQ(
+        json_validator=check_none_or(check_int), default=None
+    ),
+) -> HttpResponse:
+    if (
+        realm_deletion_in_minutes is None
+        and settings.MAX_DEACTIVATED_REALM_DELETION_DAYS is not None
+    ):
+        raise JsonableError(_("Given data deletion time is not allowed."))
+
+    if realm_deletion_in_minutes is not None and (
+        (realm_deletion_in_minutes < settings.MIN_DEACTIVATED_REALM_DELETION_DAYS * 24 * 60)
+        or (
+            settings.MAX_DEACTIVATED_REALM_DELETION_DAYS is not None
+            and realm_deletion_in_minutes > settings.MAX_DEACTIVATED_REALM_DELETION_DAYS * 24 * 60
+        )
+    ):
+        raise JsonableError(_("Given data deletion time is not allowed."))
```

Comment:
Similarly, we can be helpful and provide something like `Given data deletion time is not allowed; it must be between {__} and {___} days`